types.ts

```diff
@@ -38,9 +38,17 @@ export interface Channel<T, TClose> {
 
 /* low-level interface Which you probably will not need */
 
-export type Result<T> =
-  | { type: "resolved"; value: T }
-  | { type: "rejected"; error: Error };
+export interface OkResult<T> {
+  readonly ok: true;
+  value: T;
+}
+
+export interface ErrResult {
+  readonly ok: false;
+  error: Error;
+}
+
```

Comment:
Let's inline these like they were before. I don't think we ever want to have a reference whose type is `OkResult` or `ErrResult`. I.e. once its in the box, no peeking.